device.c

```diff
@@ -158,14 +163,17 @@ int adev_close(hw_device_t *device)
         if (xadev->xin_stream != NULL) {
             in_destroy(xadev->xin_stream);
         }
-        if (xadev->xout_stream != NULL) {
-            out_destroy(xadev->xout_stream);
+        for (i=0; i<NUMBER_OF_DEVICES_OUT; i++) {
```

Comment:
Coding style: I didn't get if you use spaces around "=" or no? Here you omit spaces and when you use "if (XX ==" you put one

Reply:
Fixed. Spaces are used inside 'for'.